include/small/rlist.h

```diff
@@ -312,12 +319,25 @@ rlist_cut_before(struct rlist *head1, struct rlist *head2, struct rlist *item)
 #define rlist_last_entry(head, type, member)				\
 	rlist_entry(rlist_last(head), type, member)
 
+/**
+ * return first entry for nonempty list and NULL otherwise
+ */
+#define rlist_last_entry_safe(head, type, member)			      \
+	(rlist_empty(head) ? NULL : rlist_last_entry((head), type, member))
+
 /**
  * return next entry
  */
 #define rlist_next_entry(item, member)					\
 	rlist_entry(rlist_next(&(item)->member), typeof(*item), member)
 
+/**
+ * return next entry if it exists or NULL if it is head
+ */
+#define rlist_next_entry_safe(item, head, member)			\
+	((rlist_next(&(item)->member) == (head)) ? NULL :               \
```

Comment:
Won't we have the same problem, we tried to fix here https://github.com/tarantool/small/commit/04ea71530aa048ff7c70a9cc4f3a052da31a8305 by introducing `rlist_entry_is_head`? We tried to avoid direct comparison with head